### ISSUE TITLE
In v5.0.6,  electron.d.ts@DesktopCapturer.getSources returnType is single object

### ISSUE THREAD

**Issue by skatpgusskat** [electron/electron#19188]
### Preflight Checklist

* [x] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/master/CONTRIBUTING.md) for this project.
* [x] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/master/CODE_OF_CONDUCT.md) that this project adheres to.
* [x] I have searched the issue tracker for an issue that matches the one I want to file, without success.

### Issue Details

* **Electron Version:**
  * 5.0.6
* **Operating System:**
  * Windows 10 17763 build
* **Last Known Working Electron version:**
  * Unknown

### Expected Behavior

the method `DesktopCapturer.getSources`'s return type must be array of `DesktopCapturerSource` like below.
```typescript
  interface DesktopCapturer extends EventEmitter {
  getSources(options: SourcesOptions, callback: (error: Error, sources: DesktopCapturerSource[]) => void): void;
    getSources(options: SourcesOptions): Promise<DesktopCapturerSource[]>; // <<< here!!
  }
```

### Actual Behavior

the method `DesktopCapturer.getSources`'s return type is single object `DesktopCapturerSource` like below.
```typescript
  interface DesktopCapturer extends EventEmitter {
  getSources(options: SourcesOptions, callback: (error: Error, sources: DesktopCapturerSource[]) => void): void;
    getSources(options: SourcesOptions): Promise<DesktopCapturerSource>; // <<< here!!
  }
```

### To Reproduce
install electorn@5.0.6 and check electron.d.ts

### Screenshots
![image](https://user-images.githubusercontent.com/3580430/60945732-5d8bb600-a327-11e9-9b6e-e1470b458e5f.png)

**Comment by skatpgusskat** [comment]
I think most of function which return array seems wrong.

For example, this `Cookies.get` method also should return `cookie[]`. Callback version do that, but not true with promise version.
![image](https://user-images.githubusercontent.com/3580430/60946398-19011a00-a329-11e9-87ba-4cd160435280.png)

**Comment by skatpgusskat** [comment]
And I test with electron master branch, they produce electron.d.ts in right way. 
![image](https://user-images.githubusercontent.com/3580430/60947042-9ed19500-a32a-11e9-8b78-fedea807069e.png)

Could you support that right electron.d.ts in 5.0.6 version too?

**Comment by MarshallOfSound** [comment]
@skatpgusskat Unfortunately not something we can easily backport / fix in earlier versions at this time.  It's not super painful to cast through to the right type, apologies though 🙇 

https://github.com/electron/electron/issues/18983#issuecomment-505694705

**Comment by skatpgusskat** [comment]
@MarshallOfSound Yes it is not super painful but the problem is 5.0.6 is current “main” version which users are using now. It’s so disappointing us. 
And it is the problem of Framework quality. Wrong type mean wrong communication in typescript. You said “it is not super painful to cast”. Seriously?

Could you one more time think about this issue? at least I suggest you let me us when this will be Ok.

**Comment by MarshallOfSound** [comment]
@skatpgusskat As I linked in that above issue, we recently rewrote our TS generation logic from the ground up.  Unfortunately that rewrite required a lot of manual editing of the documentation that can't easily be backported.  This issue is already fixed on master (in nightlies) and in the current 6.0.0 beta releases.

My point around it not being painful to cast was merely a proposed workaround while you wait to upgrade to a version of Electron with the new typescript generator, definitely not a long term / ideal solution.  It's just a thing we have to weigh up in terms of effort / maintainer time.

If you / anyone else wants to invest time in backporting https://github.com/electron/electron/pull/18103 to `5-0-x` (along with all the other docs fixes that proceeded it) I'd be more than happy to invest time in reviewing that PR, but it's not something I can prioritise right now especially when `6.0.0` is so close to stable 🙇
